pyproject.toml

```diff
@@ -20,11 +20,9 @@ dependencies = [
   "mpi4py; python_version < '3.13'",
   "h5py>3.12.1",
   "libsupermesh",
-  # NOTE: If changing the PETSc/SLEPc version then firedrake-configure also needs
-  # changing (as well as other references to petsc4py and slepc4py here)
-  "petsc4py==3.22.2",
```

Comment:
Hiding `petsc4py`/`slepc4py` dependencies into `petsctools` feels wrong, considering how important `petsc4py`/`slepc4py` are, and how likely it is that `petsctools` will only get updated once in a while (i.e., not at every upstream release)

Reply:
`petsctools` does not have petsc4py listed as a dependency in the `pyproject.toml`. I think that installing petsc4py through PyPI is not a particularly good idea as it means that we cannot assert that the PETSc and petsc4py versions match. For instance, as we just discussed on another thread, if we want to follow PETSc `main` then petsc4py cannot be included in the `pyproject.toml` because the right version is not available on PyPI. As another example, PETSc have released v3.23.0 but the corresponding petsc4py is not yet available.

`petsctools` is basically a shim package that means that petsc4py can be installed through PETSc and still found, guaranteeing that the versions match.